RFCs/0013-in-predicate-coercion.md

```diff
@@ -0,0 +1,345 @@
+incremental testing, developers might find themselves questioning why PartiQL acts in certain ways (if they have not
+fully read and understood the specification). This document will present the following scenario:
+
+> A developer wants to write a query with an IN Predicate where the RHS is an SFW sub-query. The developer might first 
+> write the SFW sub-query and run it. Then, since PartiQL supports bags as first-class citizens, the developer can copy and paste 
+> the SFW sub-query's results into the RHS of the query's IN Predicate. Then, they might shape their query around this 
+> literal (which does not perform coercion). Once their query works, they could replace the literal with their original
+> SFW sub-query -- but, then, as coercion takes place, their query would not return the same results.
+
+# Rationale and Alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The proposed design is the best option to maintain conformance with SQL. While it is limited by how PartiQL
+represents schema (through unordered structs), it allows additional semantics that are currently present in other SQL
+engines and the SQL-92 Specification.
+
+Other designs considered included:
+- Always coercing the RHS SFW sub-query elements into the list of values it holds at evaluation time. This, however,
+wouldn't work as the underlying struct (prior to coercion) cannot guarantee ordering.
+- For the scenario of a single projection item, coercing both the LHS and RHS of the IN Predicate to be a single-element
+list to make the implementation slightly easier, although this would add additional overhead during evaluation.
+- Deviating from SQL conformance and asking users to use the VALUE clause as appropriate. This would remove the drawback
+  of perceived inconsistency, but it would break PartiQL's tenet of SQL backwards compatibility.
+
+The impact of not implementing this proposal is that SQL users transitioning to PartiQL will not be able to run their SQL queries
+on PartiQL without needing to modify their existing queries. PartiQL would not be abiding by its tenet of SQL conformance.
+
+# Prior Art
+[prior-art]: #prior-art
+
+This document will examine the current PartiQL Specification and the SQL-92 Specification.
+
+#### PartiQL Specification
+
+While the PartiQL Specification does not specify rules of coercion for the elements within the result of an SFW sub-query,
+this document will reference *somewhat* relevant sections of the PartiQL Specification for the reader's knowledge.
+
+Section 9.1 of the PartiQL Specification gives insight into, specifically, how an SFW sub-query undergoes coercion into a scalar:
+
+> A SFW subquery coerces into a scalar ... if it is an SFW subquery expression that (a) is not the collection expression
+> of a *FROM* clause item and (b) is not the rhs of an *IN*. (If it is the rhs of an *IN* then it should not be coerced;
+> see note on semantics of *IN*, Section *??*).
```

Comment:
This feels like Fermat's last theorem!  ;-) 
The author suggested he had an explanation, but since he never got to writing it up anywhere, do we have to reluctantly conclude he himself, in the end, did not find the explanation convincing?  Still, due to the author's reputation, we'd love to know what was that -- even if faulty -- idea that he had in mind!

Reply:
Having mused the above, I still hold slim hope that it might be possible to contort the semantics of IN to achieve what's needed, without resorting to the RHS coercion and hopefully avoiding the perceived inconsistency drawback. (My apologies this comment is not constructive :-( )